### PR TITLE
Update tutorial to include recipe

### PR DESCRIPTION
**Pull request by rynowak**

### CODE REVIEW

File: docs/content/getting-started/first-app/_index.md [hunk 1]
```diff
@@ -108,6 +108,58 @@ Access the application by opening [http://localhost:3000](http://localhost:3000)
...
+
+This step will add a database (Redis Cache) to the application.
+
+{{% alert title="🚧🚧🚧 Under construction 🚧🚧🚧" color="warning" %}}
+TODO: turn `Recipes` in the following paragraph into a link when possible.
+{{% /alert %}}
+
+You can create a Redis Cache using [Recipes]({{< ref recipes >}}) provided by Radius. The Radius community provides Recipes for running commonly used application dependencies, including Redis.
+
+In this step you will a Recipe to:
```
Comment (AaronCrawfis) [inline docs/content/getting-started/first-app/_index.md]:
```suggestion
In this step you will use a Recipe to:
```

File: docs/content/getting-started/first-app/_index.md [hunk 2]
```diff
@@ -108,6 +108,58 @@ Access the application by opening [http://localhost:3000](http://localhost:3000)
...
+
+- Add Redis to the application.
```
Comment (AaronCrawfis) [inline docs/content/getting-started/first-app/_index.md]: Small nit: right now we're saying "The Recipe adds Redis to the application and adds environment variables to the container" which are not the Recipe. I think we should be careful to not put everything under the hat of "recipes"

File: docs/content/getting-started/first-app/_index.md [hunk 3]
```diff
@@ -108,6 +108,58 @@ Access the application by opening [http://localhost:3000](http://localhost:3000)
...
+{{< rad file="snippets/app-with-redis-snippets.bicep" embed=true marker="//REDIS" >}}
+
+The code you just added creates a Redis Cache resource, and specifies that it should be created by a recipe.
+
+Next, add this code to the container definition inside `properties`:
+
+{{< rad file="snippets/app-with-redis-snippets.bicep" embed=true marker="//CONNECTION" >}}
+
+The code you just added creates a connection between the container and the database. Based on this connection, Radius will define environment variables in the container that tell the container how to connect. You will view these in the next step.
+
```
Comment (AaronCrawfis) [inline docs/content/getting-started/first-app/_index.md:137]:
```suggestion
```

File: docs/content/getting-started/first-app/_index.md [hunk 4]
```diff
@@ -108,6 +108,58 @@ Access the application by opening [http://localhost:3000](http://localhost:3000)
...
+
+The finished `app.bicep` should like this:
+
+{{< rad file="snippets/app-with-redis.bicep" embed=true >}}
+
+## 6: Run again the application with a database
```
Comment (AaronCrawfis) [inline docs/content/getting-started/first-app/_index.md]:
```suggestion
## 6: Rerun the application with a database
```

File: docs/content/getting-started/first-app/snippets/app-with-redis.bicep [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+
+resource db 'Applications.Link/redisCaches@2022-03-15-privatepreview' = {
+  name: 'db'
+  location: 'global'
+  properties: {
+    application: application
+    environment: environment
+    mode: 'recipe'
+    recipe: {
+      name: 'redis'
```
Comment (AaronCrawfis) [inline docs/content/getting-started/first-app/snippets/app-with-redis.bicep]: Should we use a different name for the recipe here? I think the recipe name should be indicative of the use-case or the backing technology. If we're deploying lightweight containers I'd instead use "dev", "docker", or "containers"

**Comment by jasonviviano** [comment]
Have a question on what the main idea behind what the `first application` should be. I think involving recipes is always great, but wouldn't this fit better into a quickstart? And if not, I'd love to understand what the main differences would be with introducing the user to Recipes in the tutorial versus the quickstart.

File: docs/content/getting-started/first-app/snippets/app-with-redis.bicep (hunk 5 above)
Comment (rynowak) [reply docs/content/getting-started/first-app/snippets/app-with-redis.bicep]: Recipe names need to go away wholesale. It's not the right metaphor for users.

This name is based on the behaviors and limitations of how "dev-recipes" were implemented, it's not possible for me to change.

File: docs/content/getting-started/first-app/_index.md (hunk 2 above)
Comment (rynowak) [reply docs/content/getting-started/first-app/_index.md]: Lemme take a crack at rewording this section. I agree with the tension you're pointing out. I want to make sure we don't introduce more terminology.

**Comment by rynowak** [comment]
> Have a question on what the main idea behind what the `first application` should be. I think involving recipes is always great, but wouldn't this fit better into a quickstart? And if not, I'd love to understand what the main differences would be with introducing the user to Recipes in the tutorial versus the quickstart.

Hi Jason, the goal of 'first application' is to give users a "5 minutes to wow experience that highlights the most important features in Radius". I'd definitely call recipes one of the most important features.

File: docs/content/getting-started/first-app/snippets/app-with-redis-snippets.bicep [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+//REDIS
+resource db 'Applications.Link/redisCaches@2022-03-15-privatepreview' = {
+  name: 'db'
+  location: 'global'
+  properties: {
+    application: application
+    environment: environment
+    mode: 'recipe'
+    recipe: {
+      name: 'redis'
```
Comment (kachawla) [inline docs/content/getting-started/first-app/snippets/app-with-redis-snippets.bicep]: I'm assuming this is meant to use dev recipe? If so, the format is link-provider https://github.com/project-radius/radius/blob/main/pkg/corerp/frontend/controller/environments/createorupdateenvironment.go#L138

```suggestion
      name: 'redis-kubernetes'
```

Comment (rynowak) [reply docs/content/getting-started/first-app/snippets/app-with-redis-snippets.bicep]: Thanks :)

File: docs/content/getting-started/first-app/_index.md [hunk 7]
```diff
@@ -108,6 +108,54 @@ Access the application by opening [http://localhost:3000](http://localhost:3000)
...
+- Add Redis to the application using a Recipe.
+- Connect to Redis from the `demo` container using environment variables that Radius automatically sets.
+
+Open `app.bicep` in your editor and get ready to edit the file.
+
+First add some new code to `app.bicep` by pasting in the content below:
+
+{{< rad file="snippets/app-with-redis-snippets.bicep" embed=true marker="//REDIS" >}}
+
+The code you just added creates a Redis Cache resource, and specifies that it should be created by a recipe.
```
Comment (AaronCrawfis) [inline docs/content/getting-started/first-app/_index.md]:
```suggestion
The code you just added creates a Redis Cache resource and specifies that it should be created by a Recipe.
```

File: docs/content/getting-started/first-app/app.bicep [hunk 8]
```diff
@@ -0,0 +1,18 @@
+import radius as radius
```
Comment (AaronCrawfis) [inline docs/content/getting-started/first-app/app.bicep]: Should this go under snippets as well?

Comment (rynowak) [reply docs/content/getting-started/first-app/app.bicep]: It was a duplicated, deleted.

**Comment by rynowak** [comment]
@AaronCrawfis - I'm getting a build error trying to link to the recipes docs. It looks like there are multiple folders called recipes:

> Re-run Hugo with the flag --panicOnWarning to get a better error message.
ERROR 2023/02/10 19:50:17 [en] REF_NOT_FOUND: Ref "recipes": "/home/runner/work/docs/docs/docs/content/getting-started/first-app/_index.md:117:46": page reference "/getting-started/first-app/recipes" is ambiguous

How to proceed? Is there a way to link to the developer guide specifically or do I have to rename the folders?

**Comment by rynowak** [comment]
Addressed feedback, stuck on a build error.

**Comment by AaronCrawfis** [comment]
Build error will be resolved once #464 is merged

**Review by AaronCrawfis — APPROVED** [review]

**Pull request merged**